pinecone/grpc/utils.py

```diff
@@ -63,9 +64,28 @@ def parse_usage(usage: dict):
     return Usage(read_units=int(usage.get("readUnits", 0)))
 
 
-def parse_query_response(response: dict, _check_type: bool = False):
+def parse_upsert_response(response: Message, _check_type: bool = False):
+    json_response = json_format.MessageToDict(response)
+    upserted_count = json_response.get("upsertedCount", 0)
+    return UpsertResponse(upserted_count=int(upserted_count))
+
+
+def parse_update_response(response: Union[dict, Message], _check_type: bool = False):
+    return {}
+
+
+def parse_delete_response(response: Union[dict, Message], _check_type: bool = False):
+    return {}
```

Comment:
These two requests don't actually return anything, so there's nothing to do here.